src/mmw/apps/export/views.py

```diff
@@ -58,19 +58,13 @@ def hydroshare(request):
     except HydroShareResource.DoesNotExist:
         hsresource = None
 
-    # Handle missing HydroShare not finding a resource
-    if hsresource and not hs.check_resource_exists(hsresource.resource):
-        if request.method == "DELETE":
-            hsresource.delete()
-            return Response(status=status.HTTP_204_NO_CONTENT)
-
-        return Response(
-            data={'errors': ['HydroShare could not find requested resource']},
-            status=status.HTTP_404_NOT_FOUND
-        )
-
     # GET existing resource simply returns it
     if hsresource and request.method == 'GET':
+        if not hs.check_resource_exists(hsresource.resource):
+            return Response(
+                data={'errors': ['HydroShare could not find requested resource']},
```

Comment:
Think this and [this line below](https://github.com/WikiWatershed/model-my-watershed/pull/2605/commits/e2a6ba0d67f55390be8261fb88648aadf26bc3bd#diff-47f736e67321bde86fbce9a3e2236d2fR100) are too long